### DISCUSSION TITLE
nmap disallowed ?

### DISCUSSION THREAD

**Discussion by ocervell** [moby/moby#48883] [General]
I just found out today that the AppArmor docker-default security profile does not let me run nmap unless I change the binary name:
```
└─# /usr/lib/nmap/nmap
bash: /usr/lib/nmap/nmap: Operation not permitted
```

I'm looking at the [docker-default template](https://github.com/moby/moby/blob/master/profiles/apparmor/template.go) but I'm not knowledgeable enough in AppArmor to know what's blocking it in the template.

I am looking to create a new template that would allow me to run nmap while keeping the rest of the default profile.

Cheer !

**Comment by thaJeztah** [comment]
> unless I change the binary name

Hm.. interesting; I'm not an AppArmor expert either, but I recall some recent changes in Ubuntu that started to apply profiles out of band, and refuse certain operations based on those (longer discussion on https://github.com/moby/moby/pull/47749 and https://github.com/containerd/containerd/pull/10123)

What version of Docker are you running? And if you're running Ubuntu, what version of Ubuntu? Does it also make a difference what kind of image you're running? (It's possible the same applies if the container is running a newer version of Ubuntu).

**Comment by ocervell** [comment]
Running a Kali Linux (`kalli-rolling:latest`) container actually... Difficult to say the Docker version, it's running on GKE.

**Comment by ocervell** [comment]
Small update:

Running the container with `--privileged` or `--cap-add=NET_ADMIN` does work indeed, but I'm surprised by the fact that simply `cp`-ing `nmap` to another name would make it work.

Help me understand how the [default seccomp config](https://github.com/moby/moby/blob/master/profiles/seccomp/default.jso) blocks `nmap` and why copying the binary to a new name works when all the syscalls etc are the same.

**Reply by thaJeztah** [reply]
I'm not sure it's seccomp; was there something that made you conclude it's related to seccomp?

does it work for you if you run with `--security-opt seccomp=unconfined`  (without `--privileged` or `--cap-add`)?

**Reply by thaJeztah** [reply]
If it's related to AppArmor, I noticed a ticket here that mentions that installing `nmap` on the host (through snap) resulted in an AppArmor profile to be applied on the host, but not sure if it's easy to figure out what's set on the host if you don't have direct access to it; https://github.com/nmap/nmap/issues/2374#issuecomment-950546516
